frontend/src/components/smart/MerchandiseList.vue

```diff
@@ -0,0 +1,187 @@
+<template>
+  <div>
+    <div class="text-center" v-if="products.length === 0">
+      <span>{{ $t('market.merchandise.nothingToBuyAtThisTime') }}</span>
+    </div>
+    <ul class="product-grid">
+      <li class="product" v-for="product in productsForPage" :key="product.id">
+        <div class="d-flex flex-column w-100 h-100 align-items-center">
+          <img class="product-img" :src="product.thumbnail_url" alt=""/>
+          <div class="text-center font-weight-bold">
+            {{ product.name }}
+          </div>
+        </div>
+        <b-button
+          variant="primary"
+          class="shop-button"
+          :disabled="isOrderLoading"
+          @click="openChooseVariantModal(product)">
+          <span>{{ $t('market.merchandise.chooseVariant') }}</span>
+        </b-button>
+      </li>
+    </ul>
+    <b-pagination
+      v-if="products.length !== 0"
+      class="customPagination"
+      v-model="currentPage"
+      :total-rows="products.length"
+      :per-page="perPage"
+      align="center"
+    />
+    <ShippingInfoModal/>
+    <VariantChoiceModal/>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import {mapActions} from 'vuex';
+import {fromWeiEther} from '@/utils/common';
+import api from '@/api';
+import ShippingInfoModal, {Recipient} from '@/components/smart/ShippingInfoModal.vue';
+import VariantChoiceModal from '@/components/smart/VariantChoiceModal.vue';
+
+export interface Product {
+  id: number;
+  external_id: string;
+  name: string;
+  variants: number;
+  synced: number;
+  thumbnail_url: string;
+  is_ignored: boolean;
+}
+
+export interface MerchandiseOrder {
+  recipient: Recipient;
+  items: Item[];
+}
+
+export interface Item {
+  sync_variant_id: number;
+  product_id: string;
+  quantity: number;
+}
+
+interface Data {
+  products: Product[];
+  selectedProduct?: Product;
+  isLoading: boolean;
+  currentPage: number;
+  perPage: number;
+  offset: number;
+  limit: number;
+}
+
+interface StoreMappedActions {
+  getItemPrice(payload: { id: number }): Promise<number>;
+}
+
+export default Vue.extend({
+  components: {
+    ShippingInfoModal, VariantChoiceModal
+  },
+
+  data() {
+    return {
+      products: [],
+      selectedProduct: undefined,
+      isLoading: false,
+      currentPage: 1,
+      perPage: 30,
+      offset: 0,
+      limit: 100,
+    } as Data;
+  },
+
+  props: {
+    isOrderLoading: {
+      type: Boolean,
+    }
+  },
+
+  computed: {
+    rows() {
+      return this.$data.products.length;
```

Comment:
Is it necessary to use `this.$data` here instead of `this.` like you do in methods section below? I'd stick to one pattern.

Reply:
Yes, computed section needs it